lib/src/ast/property.dart

```diff
@@ -98,34 +101,88 @@ abstract class PropertyAst implements TemplateAst {
   }
 }
 
-class _ParsedPropertyAst extends TemplateAst with PropertyAst {
-  final NgToken _nameToken;
-
-  _ParsedPropertyAst(
+/// Represents a real, non-synthetic bound property assignment
+/// `[name.postfix.unit]="value"`for an element.
+///
+/// Clients should not extend, implement, or mix-in this class.
+class ParsedPropertyAst extends TemplateAst
+    with PropertyAst, OffsetInfo, SpecialOffsetInfo {
+  /// Tokens representing `[property]` attribute.
+  final NgToken prefixToken;
+  final NgToken elementDecoratorToken;
+  final NgToken suffixToken;
+
+  /// [NgAttributeValueToken] that represents `"value"`; may be `null` to
+  /// have no value.
+  final NgAttributeValueToken valueToken;
+
+  /// [NgToken] that represents the equal sign token; may be `null` to have no
+  /// value.
+  final NgToken equalSignToken;
+
+  ParsedPropertyAst(
     SourceFile sourceFile,
     NgToken beginToken,
-    this._nameToken,
-    NgToken endToken, [
-    this.expression,
+    this.prefixToken,
+    this.elementDecoratorToken,
+    this.suffixToken, [
+    this.valueToken,
+    this.equalSignToken,
   ])
-      : super.parsed(beginToken, endToken, sourceFile);
-
+      : this.expression = valueToken != null
+            ? new ExpressionAst.parse(valueToken.innerValue.lexeme,
+                sourceUrl: sourceFile.url.toString())
+            : null,
+        super.parsed(
+            beginToken,
+            valueToken == null ? suffixToken : valueToken.rightQuote,
+            sourceFile);
+
+  /// ExpressionAst of `"value"`; may be `null` to have no value.
   @override
   final ExpressionAst expression;
 
-  String get _nameWithoutBrackets {
-    return _nameToken.lexeme.substring(1, _nameToken.lexeme.length - 1);
-  }
+  String get _nameWithoutBrackets => elementDecoratorToken.lexeme;
 
+  /// Name `name` of `[name.postfix.unit]`.
   @override
   String get name => _nameWithoutBrackets.split('.').first;
 
+  /// Offset of name.
+  @override
+  int get nameOffset => elementDecoratorToken.offset;
```

Comment:
It shouldn't have to. `elementDecoratorToken` will never be null. If error recovery is disabled, the lack of identifier token will raise error immediately and exit out before even accessing this part of code. If error recovery is enabled, a synthetic is placed instead.